src/app/progress/ObjectiveDescription.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import _ from 'lodash';
+import { DestinyObjectiveDefinition } from 'bungie-api-ts/destiny2';
+import BungieImage from '../dim-ui/BungieImage';
+import { D2ManifestDefinitions } from '../destiny2/d2-definitions.service';
+import { D1ManifestDefinitions } from '../destiny1/d1-definitions.service';
+
+// matches first bracketed thing in the string, or certain private unicode characters
+const hashExtract = /^([^[\]]*)(\[[^[\]]+?\]|[\uE099-\uE154])(.*)$/u;
+
+// conversionTable
+const conversionTable = [
+  { icon: 'bow', objectiveHash: 1242546978, substring: '' },
+  { icon: 'auto_rifle', objectiveHash: 532914921, substring: '' },
+  { icon: 'pulse_rifle', objectiveHash: 2161000034, substring: '' },
+  { icon: 'scout_rifle', objectiveHash: 2062881933, substring: '' },
+  { icon: 'hand_cannon', objectiveHash: 53304862, substring: '' },
+  { icon: 'shotgun', objectiveHash: 635284441, substring: '' },
+  { icon: 'sniper_rifle', objectiveHash: 3527067345, substring: '' },
+  { icon: 'fusion_rifle', objectiveHash: 3296270292, substring: '' },
+  { icon: 'smg', objectiveHash: 2722409947, substring: '' },
+  { icon: 'rocket_launcher', objectiveHash: 2203404732, substring: '' },
+  { icon: 'sidearm', objectiveHash: 299893109, substring: '' },
+  { icon: 'melee', objectiveHash: 3951261483, substring: '' },
+  { icon: 'grenade', objectiveHash: 3711356257, substring: '' },
+  { icon: 'grenade_launcher', objectiveHash: 2152699013, substring: '' },
+  { icon: 'beam_weapon', objectiveHash: 3080184954, substring: '' },
+  { icon: 'damage_solar', objectiveHash: 2994623161, substring: '' },
+  { icon: 'damage_kinetic', objectiveHash: 3924246227, substring: '' },
+  { icon: 'headshot', objectiveHash: 3287913530, substring: '' },
+  { icon: 'damage_arc', objectiveHash: 2178780271, substring: '' },
+  { icon: 'damage_void', objectiveHash: 695106797, substring: '' },
+  { icon: 'wire_rifle', objectiveHash: 2923868331, substring: '' },
+  { icon: 'sword_heavy', objectiveHash: 989767424, substring: '' },
+  { icon: 'machinegun', objectiveHash: 1788114534, substring: '' }
+];
+
+/* supplementConversionTable: replaces icon SVG filename with the actual SVG
+ * localizes conversionTable by finding this language's replaceable substrings
+ * adds substring value for icon insertion via replacement
+ */
+function supplementConversionTable(defs) {
```

Comment:
Rather than modifying the conversion table in-place, it'd be better to return the enhanced conversion table from this function. Then your once-ed function will always return that table and you can pass it into `replaceWithIcons`, and you won't have any global state.